tests/test_wire/test_unwire.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+import magma as m
+
+
+@pytest.mark.parametrize('T', [m.Bit, m.Bits[4], m.Tuple[m.Bit, m.Bits[4]],
+                               m.Array[4, m.Bits[4]]])
+@pytest.mark.parametrize('func', [lambda x, y: x.unwire(y), m.unwire])
+def test_unwire_with_arg(T, func, caplog):
```

Comment:
* Let's write these tests without circuits, e.g.
```python
x, y, z = T(), T(), T()
...
```
* Let's use `not any(record.level >= logging.WARNING for record in caplog.records)` (`import logging` at top)
* It's probably worth saving the T and func parameterizations as variables. will reduce LOC and make it easier to write new tests if needed.